### PR TITLE
Method for stationary wall boundary assignment

### PR DESCRIPTION
**Pull request by wilove997**

### CODE REVIEW

File: pyaedt/icepak.py [hunk 1]
```diff
@@ -3182,3 +3182,447 @@ def get_face_normal(obj_face):
...
+        ext_surf_rad_material="Stainless-steel-cleaned",
+        ext_surf_rad_ref_temp="AmbientTemp",
+        ext_surf_rad_view_factor="1",
+    ):
+        """Function to assign surface wall boundary condition.
+
+        Parameters
+        ----------
+        geometry : str or int
+            name of the surface object or id of the face
```
Comment (MaxJPRey) [inline pyaedt/icepak.py:3221]:
```suggestion
            Name of the surface object or id of the face.
```

File: pyaedt/icepak.py [hunk 2]
```diff
@@ -3182,3 +3182,447 @@ def get_face_normal(obj_face):
...
+        bc_name : str
+            Name of the boundary condition. The default is ``None``.
+
```
Comment (MaxJPRey) [inline pyaedt/icepak.py:3532]:
```suggestion
```

File: pyaedt/icepak.py [hunk 3]
```diff
@@ -3182,3 +3182,447 @@ def get_face_normal(obj_face):
...
+        radiate=False,
+        radiate_surf_mat="Steel-oxidised-surface",
+        shell_conduction=False,
+    ):
+        """Function to assign surface wall boundary condition.
+
+        Parameters
+        ----------
+        geometry : str or int
+            name of the surface object or id of the face
```
Comment (MaxJPRey) [inline pyaedt/icepak.py:3529]:
```suggestion
            Name of the surface object or id of the face.
```

File: pyaedt/icepak.py [hunk 4]
```diff
@@ -3182,3 +3182,447 @@ def get_face_normal(obj_face):
...
+        radiate=False,
+        radiate_surf_mat="Steel-oxidised-surface",
+        shell_conduction=False,
+    ):
+        """Function to assign surface wall boundary condition with temperature thermal assignment.
+
+        Parameters
+        ----------
+        geometry : str or int
+            name of the surface object or id of the face
```
Comment (MaxJPRey) [inline pyaedt/icepak.py:3455]:
```suggestion
            Name of the surface object or id of the face.
```

File: pyaedt/icepak.py [hunk 5]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+            ``ext_condition="Heat Transfer Coefficient"``. The default is ``None``.
+        ref_temperature : str or float, optional
+            Reference temperature for the definition of the htc. Relevant if
+            ``ext_condition="Heat Transfer Coefficient"``.The default is ``"AmbientTemp"``.
+        material : str, optional
+            Solid material of the wall. Relevant if the thickness is non-zero. The default is ``"Al-Extruded"``.
+        radiate : bool, optional
+            Whether to enable the inner surface radiation option. The default is ``False``.
+        radiate_surf_mat : str, optional
+            Surface material used for inner surface radiation. Relevant if it is enables.
```
Comment (MaxJPRey) [inline pyaedt/icepak.py:3249]:
```suggestion
            Surface material used for inner surface radiation. Relevant if it is enabled.
```

File: pyaedt/icepak.py [hunk 6]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+        ht_correlation_free_stream_velocity="1m_per_sec",
+        ht_correlation_surface="Vertical",  # Top, Bottom, Vertical
+        ht_correlation_amb_temperature="AmbientTemp",
+        shell_conduction=False,
+        ext_surf_rad=False,
+        ext_surf_rad_material="Stainless-steel-cleaned",
+        ext_surf_rad_ref_temp="AmbientTemp",
+        ext_surf_rad_view_factor="1",
+    ):
+        """Function to assign surface wall boundary condition.
```
Comment (PipKat) [inline pyaedt/icepak.py:3216]:
```suggestion
        """Assign surface wall boundary condition.
```
Descriptions should start with the verb describing what they do.

File: pyaedt/icepak.py [hunk 7]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+
+        Parameters
+        ----------
+        geometry : str or int
+            Name of the surface object or id of the face.
```
Comment (PipKat) [inline pyaedt/icepak.py:3221]:
```suggestion
            Name of the surface object or ID of the face.
```

File: pyaedt/icepak.py [hunk 8]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+        ext_condition : str
+            Type of boundary condition. ``"Temperature"``, ``"Heat Flux"`` or ``"Heat Transfer Coefficient"``.
+        name : str
```
Comment (PipKat) [inline pyaedt/icepak.py:3224]:
```suggestion
        name : str, optional
```

File: pyaedt/icepak.py [hunk 9]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+            ``ext_condition="Heat Transfer Coefficient"``.The default is ``"AmbientTemp"``.
+        material : str, optional
+            Solid material of the wall. Relevant if the thickness is non-zero. The default is ``"Al-Extruded"``.
+        radiate : bool, optional
+            Whether to enable the inner surface radiation option. The default is ``False``.
+        radiate_surf_mat : str, optional
+            Surface material used for inner surface radiation. Relevant if it is enables.
+            The default is ``"Steel-oxidised-surface``.
+        ht_correlation : bool, optional
+            Whether to use the correlation option to compute the htc. The default is ``False``.
```
Comment (PipKat) [inline pyaedt/icepak.py:3261]:
```suggestion
            Whether to use the correlation option to compute the heat
            transfer coefficient. The default is ``False``.
```

File: pyaedt/icepak.py [hunk 10]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+        radiate_surf_mat="Steel-oxidised-surface",
+        shell_conduction=False,
+    ):
+        """Function to assign surface wall boundary condition with temperature thermal assignment.
+
+        Parameters
+        ----------
+        geometry : str or int
+            Name of the surface object or id of the face.
+        name : str
```
Comment (MaxJPRey) [inline pyaedt/icepak.py:3465]:
```suggestion
        name : str, optional
```

File: pyaedt/icepak.py [hunk 11]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+            Whether to use the external surface radiation option. Relevant if
+            ``ext_condition="Heat Transfer Coefficient"``. The default is ``False``.
+        ext_surf_rad_material : str, optional
+            Surface material for the external surface radiation option. Relevant if
+            ``ext_surf_rad=True``. The default is ``"Stainless-steel-cleaned"``.
+        ext_surf_rad_ref_temp : str or float, optional
+             Reference temperature for the external surface radiation option. Relevant if
+            ``ext_surf_rad=True``.  If a float is used, cel will be the default unit. The default is ``"AmbientTemp"``.
+        ext_surf_rad_view_factor : str or float, optional
+            View factor for the external surface radiation option. The default is ``"1"``.
```
Comment (PipKat) [inline pyaedt/icepak.py:3298]:
```suggestion
        ht_correlation_type : str, optional
            The correlation type for the heat transfer coefficient. Options are
            "Natural Convection" and "Forced Convection". This parameter is
            relevant if ``ht_correlation=True``. The default is ``"Natural Convection"``.
        ht_correlation_fluid : str, optional
            Fluid for the correlation option. This parameter is relevant if
            ``ht_correlation=True``. The default is ``"air"``.
        ht_correlation_flow_type : str, optional
            Type of flow for the correlation option. This parameter
            is relevant if ``ht_correlation=True``.  Options are ``"Turbulent"``
            and ``"Laminar"``. The default is ``"Turbulent"``.
        ht_correlation_flow_direction : str, optional
            Flow direction for the correlation option. This parameter is relevant if
            ``ht_correlation_type="Forced Convection"``. The default is ``"X"``.
        ht_correlation_value_type : str, optional
             Correlation value type for the forced convection correlation option.
             This parameter is relevant if ``ht_correlation_type="Forced Convection"``.
             Options are "Average Values" and "Local Values". The default is ``"Average Values"``.
        ht_correlation_free_stream_velocity : str or float, optional
             Free stream flow velocity. If a float value is specified, the default unit is m/s.
             This parameter is relevant if ``ht_correlation_type="Forced Convection"``.
             The default is ``"1m_per_sec"``.
        ht_correlation_surface : str, optional
            Correlation surface type for the natural convection correlation option.
            This parameter is relevant if ``ht_correlation_type="Natural Convection"``.
            Options are "Top", "Bottom", and "Vertical". The default is ``"Vertical"``.
        ht_correlation_amb_temperature : str or float, optional
            Ambient temperature for the natural convection correlation option.
            This parameter is relevant if ``ht_correlation_type="Natural Convection"``.
            If a float value is used specified, the default unit is cel.
            The default is ``"AmbientTemp"``.
        shell_conduction : bool, optional
            Whether to use the shell conduction option. The default is ``False``.
        ext_surf_rad : bool, optional
            Whether to use the external surface radiation option. This parameter
            is relevant if ``ext_condition="Heat Transfer Coefficient"``. The 
            default is ``False``.
        ext_surf_rad_material : str, optional
            Surface material for the external surface radiation option. This parameter
            is relevant if ``ext_surf_rad=True``. The default is ``"Stainless-steel-cleaned"``.
        ext_surf_rad_ref_temp : str or float, optional
             Reference temperature for the external surface radiation option. This paramter
             is relevant if  ``ext_surf_rad=True``.  If a float value is specified, the default unit is cel.
             The default is ``"AmbientTemp"``.
        ext_surf_rad_view_factor : str or float, optional
            View factor for the external surface radiation option. The default is ``"1"``.
```
For all occurrences of the following, it seems that "cel" should either be written out as Celcius or tagged as a literal string: If a float value is specified, the default unit is cel,

File: pyaedt/icepak.py [hunk 12]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+        geometry,
+        name=None,
+        heat_flux="0irrad_W_per_m2",
+        thickness="0mm",
+        material="Al-Extruded",
+        radiate=False,
+        radiate_surf_mat="Steel-oxidised-surface",
+        shell_conduction=False,
+    ):
+        """Function to assign surface wall boundary condition with heat flux thermal assignment.
```
Comment (PipKat) [inline pyaedt/icepak.py:3401]:
```suggestion
        """Assign a surface wall boundary condition with heat flux thermal assignment.
```
Can the description be made more clear? I find it confusing that you are assigning a condition to an assignment.

File: pyaedt/icepak.py [hunk 13]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+            The default is ``"0mm"``.
+        material : str, optional
+            Solid material of the wall. Relevant if the thickness is non-zero. The default is ``"Al-Extruded"``.
+        radiate : bool, optional
+            Whether to enable the inner surface radiation option. The default is ``False``.
+        radiate_surf_mat : str, optional
+            Surface material used for inner surface radiation. Relevant if it is enables.
+            The default is ``"Steel-oxidised-surface``.
+        shell_conduction : bool, optional
+            Whether to use the shell conduction option. The default is ``False``.
```
Comment (PipKat) [inline pyaedt/icepak.py:3423]:
```suggestion
        geometry : str or int
            Name of the surface object or ID of the face.
        name : str
            Name of the boundary condition. The default is ``None``.
        heat_flux : str or float, optional
            Heat flux to assign to the wall. If a float value is
            specified, the unit is irrad_W_per_m2. The default is
            ``"0irrad_W_per_m2"``.
        thickness : str or float, optional
            Thickness of the wall. If a float value is specified, the unit is the 
            current unit system set in Icepak.
            The default is ``"0mm"``.
        material : str, optional
            Solid material of the wall. This parameter is relevant if the thickness
            is non-zero. The default is ``"Al-Extruded"``.
        radiate : bool, optional
            Whether to enable the inner surface radiation option. The default is ``False``.
        radiate_surf_mat : str, optional
            Surface material for the inner surface radiation. This parameter is
            relevant if ``radiate`` is enabled. The default is ``"Steel-oxidised-surface``.
        shell_conduction : bool, optional
            Whether to use the shell conduction option. The default is ``False``.
```

File: pyaedt/icepak.py [hunk 14]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
```
Comment (PipKat) [inline pyaedt/icepak.py:3481]:
```suggestion
        geometry : str or int
            Name of the surface object or ID of the face.
        name : str, optional
            Name of the boundary condition. The default is ``None``.
        temperature : str or float, optional
            Temperature to assign to the wall. If a float value is specified,
            the unit is degrees Celsius. The default is ``"0cel"``.
        thickness : str or float, optional
            Thickness of the wall. If a float value is specified used, the unit is the
            current unit system set in Icepak. The default is ``"0mm"``.
        material : str, optional
            Solid material of the wall. This parameter is relevant if the
            thickness is a non-zero value. The default is ``"Al-Extruded"``.
        radiate : bool, optional
            Whether to enable the inner surface radiation option. The default is ``False``.
        radiate_surf_mat : str, optional
            Surface material to use for inner surface radiation. This parameter is relevant
            if ``radiate`` is enabled. The default is ``"Steel-oxidised-surface``.
        shell_conduction : bool, optional
            Whether to use the shell conduction option. The default is ``False``.
```

File: pyaedt/icepak.py [hunk 15]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+        ht_correlation_amb_temperature="AmbientTemp",
+        ext_surf_rad=False,
+        ext_surf_rad_material="Stainless-steel-cleaned",
+        ext_surf_rad_ref_temp="AmbientTemp",
+        ext_surf_rad_view_factor="1",
+        radiate=False,
+        radiate_surf_mat="Steel-oxidised-surface",
+        shell_conduction=False,
+    ):
+        """Function to assign surface wall boundary condition.
```
Comment (PipKat) [inline pyaedt/icepak.py:3533]:
```suggestion
        """Assign a surface wall boundary condition.
```

File: pyaedt/icepak.py [hunk 16]
```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
...
+            Whether to use the external surface radiation option. Relevant if
+            ``ext_condition="Heat Transfer Coefficient"``. The default is ``False``.
+        ext_surf_rad_material : str, optional
+            Surface material for the external surface radiation option. Relevant if
+            ``ext_surf_rad=True``. The default is ``"Stainless-steel-cleaned"``.
+        ext_surf_rad_ref_temp : str or float, optional
+             Reference temperature for the external surface radiation option. Relevant if
+            ``ext_surf_rad=True``.  If a float is used, cel will be the default unit. The default is ``"AmbientTemp"``.
+        ext_surf_rad_view_factor : str or float, optional
+            View factor for the external surface radiation option. The default is ``"1"``.
```
Comment (PipKat) [inline pyaedt/icepak.py:3598]:
```suggestion
        name : str
            Name of the boundary condition. The default is ``None``.
        htc : str or float, optional
            Heat transfer coefficient to assign to the wall. If a float value
            is specified, the unit is w_per_m2kel. The default is ``"0w_per_m2kel"``.
        thickness : str or float, optional
            Thickness of the wall. If a float value is specified, the unit is the
            current unit system set in Icepak. The default is ``"0mm"``.
        htc_dataset : str, optional
            Dataset that represents the dependency of the heat transfer
            coefficient on temperature. This parameter is relevant if
            ``ext_condition="Heat Transfer Coefficient"``. The default is ``None``.
        ref_temperature : str or float, optional
            Reference temperature for the definition of the heat transfer
            coefficient. This parameter is relevant if
            ``ext_condition="Heat Transfer Coefficient"``. The default is ``"AmbientTemp"``.
        material : str, optional
            Solid material of the wall. This parameter is relevant if the thickness
            is non-zero. The default is ``"Al-Extruded"``.
        radiate : bool, optional
            Whether to enable the inner surface radiation option. The default is ``False``.
        radiate_surf_mat : str, optional
            Surface material for inner surface radiation. This parameter is relevant
            if ``radiate`` is enabled. The default is ``"Steel-oxidised-surface``.
        ht_correlation : bool, optional
            Whether to use the correlation option to compute the heat transfer
            coefficient. The default is ``False``.
        ht_correlation_type : str, optional
            Correlation type for the correlation option. This parameter is
            relevant if ``ht_correlation=True``. Options are "Natural Convection"
            and "Forced Convection". The default is ``"Natural Convection"``.
        ht_correlation_fluid : str, optional
            Fluid for the correlation option. This parameter is relevant if
            ``ht_correlation=True``. The default is ``"air"``.
        ht_correlation_flow_type : str, optional
            Type of flow for the correlation option. This parameter is relevant
            if ``ht_correlation=True``. Options are ``"Turbulent"`` and ``"Laminar"``.
            The default is ``"Turbulent"``.
        ht_correlation_flow_direction : str, optional
            Flow direction for the correlation option. This parameter is relevant
            if ``ht_correlation_type="Forced Convection"``. The default is ``"X"``.
        ht_correlation_value_type : str, optional
             Value type for the forced convection correlation option. This
             parameter is relevant if ``ht_correlation_type="Forced Convection"``.
             Options are "Average Values" and "Local Values". The default
             is ``"Average Values"``.
        ht_correlation_free_stream_velocity : str or float, optional
             Free stream flow velocity. This parameter is relevant if
             ``ht_correlation_type="Forced Convection"``.  If a float
             value is specified, m/s is the default unit. The default is ``"1m_per_sec"``.
        ht_correlation_surface : str, optional
            Surface for the natural convection correlation option. This parameter is
            relevant if ``ht_correlation_type="Natural Convection"``. Options are "Top",
            "Bottom", and "Vertical". The default is ``"Vertical"``.
        ht_correlation_amb_temperature : str or float, optional
            Ambient temperature for the natural convection correlation option.
            This parameter is relevant if ``ht_correlation_type="Natural Convection"``.
            If a float value is specified, the default unit is cel. The default is
            ``"AmbientTemp"``.
        shell_conduction : bool, optional
            Whether to use the shell conduction option. The default is ``False``.
        ext_surf_rad : bool, optional
            Whether to use the external surface radiation option. This parameter
            is relevant if ``ext_condition="Heat Transfer Coefficient"``. The default is ``False``.
        ext_surf_rad_material : str, optional
            Surface material for the external surface radiation option. This parameter is
            relevant if ``ext_surf_rad=True``. The default is ``"Stainless-steel-cleaned"``.
        ext_surf_rad_ref_temp : str or float, optional
             Reference temperature for the external surface radiation option. This
             parameter is relevant if ``ext_surf_rad=True``.  If a float value is specified, the
             default unit is cel. The default is ``"AmbientTemp"``.
        ext_surf_rad_view_factor : str or float, optional
            View factor for the external surface radiation option. The default is ``"1"``.
```

**Review by PipKat — APPROVED** [review]
I did notice that "oxidised" was used in an option choice rather than the American spelling "oxidized" and thought I'd point this out--though I understand it's just matching the options in the software.

**Comment by PipKat** [comment]
@MaxJPRey @wilove997 I made a whole bunch of editorial review comments on this PR and somehow it got merged before these comments were incorporated. Is there any way to get these edits into the main branch?

**Comment by MaxJPRey** [comment]
No problem @PipKat. The "auto-merged" option was activated, so when you approved the PR, it was merged automatically.
@wilove997 Can you incorporate Kathy's suggestion in a new PR?
I am creating a new issue so we don't forget about it.

**Comment by MaxJPRey** [comment]
@wilove997 @PipKat I opened #2303 . Thanks.

**Comment by wilove997** [comment]
@MaxJPRey @PipKat I'm sorry. When I enabled the auto-merge I thought those were the last suggestions.

**Comment by MaxJPRey** [comment]
No worries @wilove997 .

File: pyaedt/icepak.py (hunk 12 above)
Comment (wilove997) [reply pyaedt/icepak.py:3401]: In all these descriptions, I'll use:

"Assign a surface wall boundary condition with specified ......"

This should be clearer.

**Comment by wilove997** [comment]
@PipKat @MaxJPRey Since many comments involved shorter lines in docstrings, is there a rule on their length I should follow in future PRs?

**Comment by PipKat** [comment]
@wilove997  @MaxJPRey The goal is for the description to fit on one line on the API reference page for the class in the generated doc. Note that you can always insert a blank line and then provide as much additional information as you want. Just keep in mind that we want this additional information to be provided in complete, readable sentences. Does this help? @MaxJPRey might know how many characters can actually fit in that line, but I don't.

**Comment by MaxJPRey** [comment]
@wilove997 @PipKat the maximum line length for PyAEDT is 120.

**Comment by PipKat** [comment]
> @wilove997 @PipKat the maximum line length for PyAEDT is 120 characters.

@MaxJPRey 
So, 120 characters will ensure that the description appears on one line here? Alex wanted the descriptions short so that the summary tables don't truncate the descriptions. I just wanted to confirm.

image

**Comment by MaxJPRey** [comment]
@PipKat 
So for the comment to appear on a single line, we can have up to 100 characters from what I can see in the documentation.
Which is different from the maximum line length of the source code: 120 characters.
Thanks @PipKat for pointing out the disconnect between doc and source code.

**Comment by wilove997** [comment]
Thank you both for the discussion @PipKat @MaxJPRey . I didn't know about this limit before today. I'll keep that in mind for the future :)

**Pull request merged**